### DISCUSSION TITLE
04. What is the true cost of implementing open source solutions into clinical data analytic processes?

### DISCUSSION THREAD

**Discussion by MichaelRimler** [phuse-org/OSTCDA#4] [General]
What is the true cost of implementing open source solutions into clinical data analytic processes?

**Comment by andyofsmeg** [comment]
I've lost a lot of my hair over the past 6 years.

**Comment by andyofsmeg** [comment]
Obviously all offset against productivity gains, license saving, etc:

Time.  Starting today, I reckon 3-5 years of disruption.  4-5 years before you're better off.
Licensing - commercial products (eg Posit suite) that are generally required to support R ecosystems (else the cost of DIY).  These are still much cheaper than the closed source alternatives, but companies will be paying for both for years.  
Churn.  Specifically a loss of domain expertise from senior staff if existing solutions are decommissioned.
Training - direct costs + reduced productivity whilst learning

**Comment by DrLynTaylor** [comment]
If we talk about the historical cost of recruiting people with SAS programming experience and ongoing recruitment difficulties, perhaps this would help to balance the scales on cost.   Opening our industry up to employing experienced open source programmers would hopefully widen the pool of talent and would also reduce new staffs onboarding time (no need for 3 months SAS training !).

**Comment by manciniedoardo** [comment]
Connected to the comments from @andyofsmeg and @DrLynTaylor regarding time/training/recruiting here is a also an implicit "people" cost - companies may end up losing experienced SAS programmers who are not willing or able to make the switch to open source solutions, but who would've otherwise still remained useful assets.

**Comment by CraigWangStat** [comment]
We cannot speak cost without benefit, I think we need the same metric for cost and benefit to make convincing arguments to different stakeholders.

**Comment by MikeKSmith-Pfizer** [comment]
R is free, as in beer (no cost). The R packages are "free" as in beer and in speech (we can use them free of cost and they are also open for us to contribute to). But standing up a professional environment for deploying R in a validated way (accurate, reproducible, traceable) takes effort, resource and probably $$$ if you want to implement it well. Depends if you have internal resource to manage the infrastructure - but even internal resource tends not to be "free" (no cost). 

So cost saving over proprietary systems can be made, but it doesn't _**eliminate**_ cost.
